tests/unit/plugins/test_writer.py

```diff
@@ -0,0 +1,138 @@
+import pathlib
+import shutil
+import tempfile
+
+import pytest
+
+from lineapy.plugins.pipeline_writers import generate_pipeline_files
+from lineapy.utils.utils import get_system_python_version, prettify
+
+
+def check_requirements_txt(t1: str, t2: str):
+    return set(t1.split("\n")) == set(t2.split("\n"))
+
+
+@pytest.mark.parametrize(
+    "input_script1, input_script2, artifact_list, framework, pipeline_name, dependencies, dag_config",
+    [
+        pytest.param(
+            "simple",
+            "complex",
+            ["a0", "b0"],
+            "SCRIPT",
+            "script_pipeline_a0_b0",
+            {},
+            {},
+            id="script_pipeline_a0_b0",
+        ),
+        pytest.param(
+            "simple",
+            "complex",
+            ["a0", "b0"],
+            "AIRFLOW",
+            "airflow_pipeline_a0_b0",
+            {},
+            {},
+            id="airflow_pipeline_a0_b0",
+        ),
+        pytest.param(
+            "simple",
+            "complex",
+            ["a0", "b0"],
+            "SCRIPT",
+            "script_pipeline_a0_b0_dependencies",
+            {"a0": {"b0"}},
+            {},
+            id="script_pipeline_a0_b0_dependencies",
+        ),
+        pytest.param(
+            "simple",
+            "complex",
+            ["a0", "b0"],
+            "AIRFLOW",
+            "airflow_pipeline_a0_b0_dependencies",
+            {"a0": {"b0"}},
+            {},
+            id="airflow_pipeline_a0_b0_dependencies",
+        ),
+        pytest.param(
+            "housing",
+            "",
+            ["y", "p value"],
+            "SCRIPT",
+            "script_pipeline_housing_w_dependencies",
+            {"p value": {"y"}},
+            {},
+            id="script_pipeline_housing_w_dependencies",
+        ),
+    ],
+)
+def test_pipeline_generation(
+    execute,
+    input_script1,
+    input_script2,
+    artifact_list,
+    framework,
+    pipeline_name,
+    dependencies,
+    dag_config,
+):
+    """
+    Test two sessions
+    """
+
+    code1 = pathlib.Path(
+        "tests/unit/graph_reader/inputs/" + input_script1
```

Comment:
Using string literals and string concatenation to construct paths is dangerous. Prefer 
`Path("tests") / "unit" / "graph_reader" / "inputs" / input_script1`

Reply:
Awesome, thanks for pointing this out! Using another equivalent way, i.e., `Path("tests", "unit", "graph_reader", "inputs", input_script1)`